src/agent.c

```diff
@@ -396,6 +406,295 @@ struct agent_ops agent_ops_pageant = {
     agent_transact_pageant,
     agent_disconnect_pageant
 };
+
+/* Code to talk to OpenSSH was taken and modified from the Win32 port of
```

Comment:
I would like to suggest to move the copied Windows specific code to it's own dedicated source file, e.g. "agent_win.c" maybe?

Reply:
Should the same be done for the Putty-specific code as that is in a similar situation?

Reply:
Good idea for another PR. For now focus on what was taken from OpenSSH for Windows and please also show path and commit revision which was copied

Reply:
I've noted the commit SHA, git repo, and the main files that the implementation was based on. Does this fit with your expectations?

Reply:
@mback2k Do the changes made address your concerns in a way you are happy with?

Reply:
Thanks again, see new comments below.